nmstatectl/nmstatectl.py

```diff
@@ -394,3 +396,15 @@ def _filter_routes(state, patterns):
                 if fnmatch.fnmatch(route[Route.NEXT_HOP_INTERFACE], pattern):
                     routes[route_type].append(route)
     return routes
+
+
+def _filter_route_rule(state, patterns):
+
+    route_rules = {RouteRule.CONFIG: []}
+
+    for rule in state.get(RouteRule.CONFIG, {}).get(RouteRule.ROUTE_TABLE, {}):
+        for pattern in patterns:
+            if fnmatch.fnmatch(rule[Route.NEXT_HOP_INTERFACE], pattern):
```

Comment:
The rule rule does not have `Route.NEXT_HOP_INTERFACE`.

A suggestion on the approach. The `_filter_routes()` has got all interested routes, you just need to search whether
any rule is pointing to the route table of those routes.

Reply:
@cathay4t 
I have made the changes so that route rule filters rule which match the table id in filtered route.